tests/Doctrine/Tests/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -61,22 +64,22 @@ public function testGetParentMetadata(): void
     public function testGetCachedMetadata(): void
     {
         $metadata = $this->createMock(ClassMetadata::class);
-        $cache    = new ArrayCache();
+        $cache    = $this->getArrayCache();
         $cache->save(ChildEntity::class . '$CLASSMETADATA', $metadata);
 
         $this->cmf->setCacheDriver($cache);
 
-        self::assertSame($metadata, $this->cmf->getMetadataFor(ChildEntity::class));
+        self::assertEquals($metadata, $this->cmf->getMetadataFor(ChildEntity::class));
```

Comment:
💡 Those tests were actually broken. A cache usually serialized/deserializes the given object, so you cannot expect the receive the very same instance you've inserted.